Console/Command/ReplicaSyncCommand.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Console\Command;
+
+use Algolia\AlgoliaSearch\Api\Console\ReplicaSyncCommandInterface;
+use Algolia\AlgoliaSearch\Api\Product\ReplicaManagerInterface;
+use Algolia\AlgoliaSearch\Exception\ReplicaLimitExceededException;
+use Algolia\AlgoliaSearch\Exceptions\AlgoliaException;
+use Algolia\AlgoliaSearch\Exceptions\BadRequestException;
+use Algolia\AlgoliaSearch\Exceptions\ExceededRetriesException;
+use Algolia\AlgoliaSearch\Helper\Entity\ProductHelper;
+use Algolia\AlgoliaSearch\Service\StoreNameFetcher;
+use Magento\Framework\App\State;
+use Magento\Framework\Console\Cli;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\StoreManagerInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Algolia\AlgoliaSearch\Console\Traits\ReplicaSyncCommandTrait;
+
+class ReplicaSyncCommand extends AbstractReplicaCommand implements ReplicaSyncCommandInterface
+{
+    use ReplicaSyncCommandTrait;
+
+    public function __construct(
+
+        protected ProductHelper           $productHelper,
+        protected ReplicaManagerInterface $replicaManager,
+        protected StoreManagerInterface   $storeManager,
+        protected StoreNameFetcher        $storeNameFetcher,
+        State                             $state,
+        ?string                           $name = null
+    )
+    {
+        parent::__construct($state, $name);
+    }
+
+    protected function getReplicaCommandName(): string
+    {
+        return 'sync';
+    }
+
+    protected function getStoreArgumentDescription(): string
+    {
+        return 'ID(s) for store(s) to be synced with Algolia (optional), if not specified all stores will be synced';
+    }
+
+    protected function getCommandDescription(): string
+    {
+        return 'Sync configured sorting attributes in Magento to Algolia replica indices';
```

Comment:
It is the same function as the admin but this essentially splits out the more useful "rebuild" command for granularity and ease of testing and debugging. 

It might be a real edge case but I also could envision this invoked from another shell script or cron job, perhaps if there were some kind of external service altering the state of the product data or the sorting configuration outside of the Magento admin. In general I could see this and other future CLIs being useful for such a purpose. 

But in most cases, these won't be needed but only when diagnosing or troubleshooting the replica state. I was planning on adding a section in our Algolia docs to provide more guidance here on when you might use these.

Reply:
Updated to clarify the intent of the `rebuild` vs regular `sync` command
![image](https://github.com/algolia/algoliasearch-magento-2/assets/986313/6c9cb3de-da14-49b6-9352-b18982af4f80)